src/filechooser.c

```diff
@@ -652,6 +655,11 @@ handle_open (XdpImplFileChooser *object,
 
   gtk_widget_show (dialog);
 
+#ifdef HAVE_GTK_X11
+  if (GDK_IS_X11_WINDOW (gtk_widget_get_window (dialog)))
+    gtk_window_present_with_time (GTK_WINDOW (dialog), GDK_CURRENT_TIME);
```

Comment:
Does this really need an x11 guard?

Reply:
*Technically* no, but afaik otherwise it would be, at best, a waste of cycles. (I'm not familiar enough with how this works on Wayland to say what it would do at worst.)

Reply:
This just adds complexity. If the wayland backend is a noop thats fine.